index.js

```diff
@@ -2,6 +2,16 @@
 var _ = require('lodash');
 
 module.exports = function (inputObj, opts) {
+  // Throw error if `inputObj` does not exist or is not an array/object.
+  if ( !inputObj || typeof inputObj !== 'object'  ) {
```

Comment:
This code looks odd, better to just check undefined. We need an object, anything else will break the module anyway. So, just the `typeof` check should be enough. Let me know if there is some other case you had in mind when you wrote this. :slightly_smiling_face:

Reply:
I was thinking that the module would error out immediately if/when invoked with an invalid value (eg. anything other than an array of object). 

- `typeof inputObj === 'undefined` would ensure that an argument is present, but not that it's of the correct type.
- `typeof inputObj === 'object` would ensure that the argument is either an object or array, but a `null` value would also be let through.
- The leading `!inputObj` was added to protect against `null` values.

Maybe there's a more elegant solution, or maybe it's too strict? I'm very happy to make any changes you request, so let me know what you think>

Reply:
UPDATE: This is addressed by your proposed update in the comment(s) below.